flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java

```diff
@@ -114,12 +116,23 @@
                     .enumType(NotNullEnforcer.class)
                     .defaultValue(NotNullEnforcer.ERROR)
                     .withDescription(
-                            "The NOT NULL column constraint on a table enforces that "
-                                    + "null values can't be inserted into the table. Flink supports "
-                                    + "'error' (default) and 'drop' enforcement behavior. By default, "
-                                    + "Flink will check values and throw runtime exception when null values writing "
-                                    + "into NOT NULL columns. Users can change the behavior to 'drop' to "
-                                    + "silently drop such records without throwing exception.");
+                            "Determines whether the NOT NULL column constraint on a table enforces that "
+                                    + "null values can't be inserted into the table. By default, "
+                                    + "Flink will check values and throw runtime exception when attempting "
+                                    + "to write null values writing into NOT NULL columns. Users can change "
+                                    + "the behavior to silently drop such records without throwing exception.");
+
+    @Documentation.TableOption(execMode = Documentation.ExecMode.BATCH_STREAMING)
+    public static final ConfigOption<CharPrecisionEnforcer>
+            TABLE_EXEC_SINK_CHAR_PRECISION_ENFORCER =
+                    key("table.exec.sink.char-precision-enforcer")
+                            .enumType(CharPrecisionEnforcer.class)
+                            .defaultValue(CharPrecisionEnforcer.IGNORE)
+                            .withDescription(
+                                    "Determines whether string values for columns with CHAR(<precision>)/VARCHAR(<precision>) "
+                                            + "types will be trimmed, so that their length will match the one defined by the "
+                                            + "precision of their respective CHAR/VARCHAR column type. By default, no trimming "
+                                            + "is applied.");
```

Comment:
The default value is already documented.

```suggestion
                                            + "precision of their respective CHAR/VARCHAR column type.");
```